controllers/azurecluster_reconciler.go

```diff
@@ -143,6 +143,10 @@ func (s *azureClusterService) getService(name string) (azure.ServiceReconciler,
 // setFailureDomainsForLocation sets the AzureCluster Status failure domains based on which Azure Availability Zones are available in the cluster location.
 // Note that this is not done in a webhook as it requires API calls to fetch the availability zones.
 func (s *azureClusterService) setFailureDomainsForLocation(ctx context.Context) error {
+	if s.scope.ExtendedLocation() != nil && s.scope.ExtendedLocation().Name != "" {
+		return nil
+	}
+
```

Comment:
Is it possible to have ExtendedLocation without name? If not, then the second condition is not required.

Reply:
Only ExtendedLocation instance with name and type can go through the verification and go into reconciler. So will remove the second condition.